realm/realm-library/src/androidTest/java/io/realm/RxJavaTests.java

```diff
@@ -247,8 +247,17 @@ public void findFirstAsync_emittedOnSubscribe() {
         final AllTypes asyncObj = realm.where(AllTypes.class).findFirstAsync();
         subscription = asyncObj.<AllTypes>asFlowable().subscribe(rxObject -> {
             assertTrue(rxObject.isFrozen());
+            // Because the subscription is run asynchronously. There is a chance
+            // the query resolved before the subscription triggers.
+            // This means it is not deterministic what state is first emitted here.
+            // It can either be a fully loaded object or one that is still loading.
             assertFalse(rxObject.isValid());
-            assertFalse(rxObject.isLoaded());
+            if (rxObject.isLoaded()) {
```

Comment:
Is it not possible to use Realm.refresh() here?

Reply:
Yes, that would actually also have worked. Might make for a cleaner test 👍

Reply:
Ups, to fast there. `refresh()` doesn't work on frozen objects which we emit as the default now.

Reply:
Maybe `realmObject.load()`? :thinking:

Reply:
No, that doesn't work either. It isn't possible to change the state of frozen objects.
The only thing we can do is not emitting unloaded objects, which has to be filtered in the `RealmObservableFactory`.

The issue is that both `findFirstAsync` and `subscribe` are now asynchronous. Previously, `subscribe` was synchronous so you always knew that the first emission was an unloaded object.